apps/web/src/components/milestone-logs/recruitment/Record.tsx

```diff
@@ -145,11 +146,16 @@ export const Record: React.FC<RecordProps> = ({ job, update }) => {
               />
             ))}
             <AddMilestone
-              applicantId={applicantId as string}
+              applicantId={applicantId}
               job={job}
               setJobMilestones={handleNewMilestones}
             />
-            <AddRecordModal job={job} onClose={handleModalClose} visible={modalVisible} />
+            <AddRecordModal
+              job={job}
+              milestones={milestones}
```

Comment:
I don't like this approach but still no better idea.
Job has milestone data but it could be stale when editing milestones.